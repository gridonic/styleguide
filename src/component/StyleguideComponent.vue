<template>
    <div class="styleguide-component">
        <h2 class="name">
            {{ name }}
        </h2>
        <div class="examples">
            <h3 class="title">
                Example
            </h3>
            <div
                v-for="example in examples"
                :key="example"
            >
                <!-- eslint-disable-next-line -->
                <span v-html="example"/>
            </div>
        </div>
    </div>
</template>

<script>
export default {
    props: {
        name: {
            type: String,
            required: true,
        },
        examples: {
            type: Array,
            default: () => [],
        },
    },
};
</script>

<style lang="scss" scoped>
    .styleguide-component {
        > .name {
            margin: 0;
        }

        > .examples {
            > .title {
                text-transform: uppercase;
                color: #E7E3D7;
                font-weight: normal;
            }
        }
    }
</style>
