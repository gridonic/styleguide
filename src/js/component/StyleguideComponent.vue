<template>
    <div class="styleguide-component">
        <div class="information -typography-styleguide">
            <h2 class="name">
                {{ name }}
            </h2>
        </div>
        <div class="exampletitle -typography-styleguide">
            <h3>
                Example
            </h3>
        </div>
        <div class="examples">
            <div
                class="example"
                v-for="example in examples"
                :key="example"
            >
                <!-- eslint-disable-next-line -->
                <span v-html="example"/>
            </div>
        </div>
    </div>
</template>

<script>
export default {
    props: {
        name: {
            type: String,
            required: true,
        },
        examples: {
            type: Array,
            default: () => [],
        },
    },
};
</script>

<style lang="scss" scoped>
    .styleguide-component {
        > .information {
            > .name {
                margin: 0;
            }
        }

        > .exampletitle {
            margin-top: 20px;
        }

        > .examples {
            margin-top: 10px;

            display: grid;
            grid-auto-flow: row dense;
            grid-gap: 15px;
        }
    }
</style>
