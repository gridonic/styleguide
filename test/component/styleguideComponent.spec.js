import { shallowMount } from '@vue/test-utils';

import StyleguideComponent from '@/js/component/StyleguideComponent.vue';

describe('StyleguideComponent', () => {
    const build = () => {
        const wrapper = shallowMount(StyleguideComponent, {
            propsData: {
                name: 'Fancy Component',
            },
        });

        return {
            wrapper,
            name: () => wrapper.find('.name'),
        };
    };

    it('renders the component', () => {
        const { wrapper } = build();

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders the components name', () => {
        const { name } = build();

        expect(name().exists()).toBeTruthy();
        expect(name().text()).toBe('Fancy Component');
    });
});
