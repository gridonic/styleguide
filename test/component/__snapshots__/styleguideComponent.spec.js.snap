// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleguideComponent renders the component 1`] = `
"<div class=\\"styleguide-component\\"><h2 class=\\"name\\">
        Fancy Component
    </h2> <div class=\\"examples\\"><h3 class=\\"title\\">
            Example
        </h3> </div></div>"
`;
